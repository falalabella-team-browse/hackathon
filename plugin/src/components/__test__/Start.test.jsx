import React from "react";
import { shallow } from "enzyme";
import Star from "../Star";

describe("Star", () => {
  it("should render properly, with default props", () => {
    const wrapper = shallow(<Star />);
    expect(wrapper).toMatchSnapshot();
  });

  it("should render properly, with some props", () => {
    const wrapper = shallow(
      <Star width={25} height={25} view="EMPTY" gap={0} />
    );
    wrapper.find("svg").simulate("click");
    expect(wrapper).toMatchSnapshot();
  });

  it("should be clickable", () => {
    const fn = jest.fn();
    const wrapper = shallow(
      <Star width={25} height={25} view="HALF" onClick={fn} gap={0} />
    );
    wrapper.find("svg").simulate("click");
    expect(fn).toBeCalled();
  });
});
