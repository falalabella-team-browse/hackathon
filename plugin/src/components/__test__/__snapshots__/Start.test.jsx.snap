// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Star should render properly, with default props 1`] = `ShallowWrapper {}`;

exports[`Star should render properly, with some props 1`] = `ShallowWrapper {}`;
